src/masternodes/validation.cpp

```diff
@@ -2483,9 +2484,31 @@ static Res ProcessEVMQueue(const CBlock &block, const CBlockIndex *pindex, CCust
     res = cache.AddBalance(minerAddress, {DCT_ID{}, static_cast<CAmount>(blockResult.total_priority_fees)});
     if (!res) return res;
 
-    return Res::Ok();
+    return {blockResult.block_number, Res::Ok()};
+}
+
+
+static Res ProcessDST20Migration(const CBlock &block, const CBlockIndex *pindex, CCustomCSView &cache, const CChainParams& chainparams, const uint64_t evmQueueId, std::array<uint8_t, 20>& beneficiary, const bool evmEnabledOnBlockHead) {
+    if (!IsEVMEnabled(pindex->nHeight, cache, chainparams.GetConsensus())) return Res::Ok();
+    auto time = GetTimeMillis();
+    LogPrintf("DST20 migration ...\n");
+
+    CrossBoundaryResult result;
+    cache.ForEachLoanToken([&](DCT_ID const & key, CLoanView::CLoanSetLoanTokenImpl loanToken) {
```

Comment:
Should also happen for "normal" DATs, right? Or is the remaining part planned for another PR?